doc/build/dts/design.rst

```diff
@@ -50,36 +50,9 @@ Source compatibility with other operating systems
 *************************************************
 
 Zephyr's devicetree tooling is based on a generic layer which is interoperable
-with other devicetree users, such as the Linux kernel.
+at a DTS syntax level with other devicetree users, such as the Linux kernel.
 
-Zephyr's binding language *semantics* can support Zephyr-specific attributes,
-but shall not express Zephyr-specific relationships.
-
-Examples
-========
-
-- Zephyr's devicetree source parser, :ref:`dtlib.py <dt-scripts>`, is
-  source-compatible with other tools like `dtc`_ in both directions:
-  :file:`dtlib.py` can parse ``dtc`` output, and ``dtc`` can parse
-  :file:`dtlib.py` output.
-
-- Zephyr's "extended dtlib" library, :file:`edtlib.py`, shall not include
-  Zephyr-specific features. Its purpose is to provide a higher-level view of the
-  devicetree for common elements like interrupts and buses.
-
-  Only the high-level :file:`gen_defines.py` script, which is built on top of
-  :file:`edtlib.py`, contains Zephyr-specific knowledge and features.
-
-.. _dtc: https://git.kernel.org/pub/scm/utils/dtc/dtc.git/about/
-
-Example remaining work
-======================
-
-- Zephyr has a custom :ref:`dt-bindings` language *syntax*. While Linux's
-  dtschema does not yet meet Zephyr's needs, we should try to follow what it is
-  capable of representing in Zephyr's own bindings.
-
-- Due to inflexibility in the bindings language, Zephyr cannot support the full
-  set of bindings supported by Linux.
-
-- Devicetree source sharing between Zephyr and Linux is not done.
+Zephyr's devicetree bindings should be compatible with Linux bindings as well
+whenever that is practical. However, the needs of the two operating systems are
+different, so Zephyr bindings can be incompatible with Linux provided there is
+a strong enough justification to convince the devicetree maintainers.
```

Comment:
Taking the liberty to comment on a draft PR here, but I don't think this should be solely left up to the devicetree maintainers. The maintainers of the driver class in question should also have a say in this.

Reply:
Yep